popgym/envs/count_recall.py

```diff
@@ -62,26 +64,37 @@ def render(self):
         queried = self.deck_type[self.query]
         print(f"Dealt {dealt}, recall {queried} count")
 
+    def get_state(self):
+        state = (
+            (self.counts / self.max_card_count).astype(np.float32),
+            (self.query_counts / self.max_card_count).astype(np.float32),
+            self.last_obs.copy(),
+        )
+        return state
+
     def step(self, action):
-        done = False
-        action = action.item()
+        if isinstance(action, np.ndarray):
+            action = action.item()
 
         self.prev_query = self.query
         prev_count = self.counts[self.prev_query]
         self.value, self.query = self.deal()
         self.counts[self.value] += 1
+        self.query_counts[self.query] += 1
 
-        error = abs(prev_count - action)
-        clamped = min(error, self.error_clamp)
-        reward = self.reward_scale * (1 - 2 * clamped / self.error_clamp)
+        reward = 1 if action == prev_count else -1
+        reward *= self.reward_scale
```

Comment:
Are policies still able to learn using the discrete action space? It seems like it might be harder to learn up to ~52 discrete outputs rather than a single float.